airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -95,11 +96,13 @@ public class GlobalAsyncStateManager {
   private long retroactiveGlobalStateId = 0;
   // All access to this field MUST be guarded by a synchronized(lock) block
   private long arrivalNumber = 0;
+  private final Consumer<AirbyteMessage> outputRecordCollector;
 
-  private final Object LOCK = new Object();
+  private static final Object LOCK = new Object();
 
-  public GlobalAsyncStateManager(final GlobalMemoryManager memoryManager) {
+  public GlobalAsyncStateManager(final GlobalMemoryManager memoryManager, final Consumer<AirbyteMessage> outputRecordCollector) {
     this.memoryManager = memoryManager;
+    this.outputRecordCollector = outputRecordCollector;
```

Comment:
Can we avoid holding a copy of outputRecordCollector at instance level and use only in flushStates ? This reduces the possibility of future abuse of this instance variable.